### PR TITLE
Redirect signin

### LINKED ISSUE #29: Direct user directly to custom sign in page
Currently, in order to get to our custom sign in page, the user needs to click this sign in button.
Screen Shot 2022-06-24 at 4 53 19 PM
 
The purpose of this ticket is to remove this step so that when the user goes to the website, they are automatically directed to the custom sign in page. We can largely reference the app portal code for this ticket!

- [ ] Create a folder + file /server/getServerSideSessionOrRedirect and copy over the contents of /server/getServerSideSessionOrRedirect in the app portal to this file in our repo
- [ ] Similar to how the function is called in the app portal index.tsx page, call it in all of our pages to redirect the user to the auth/signin page. 
- [ ] Remove the signin button and the 'not signed in text' from all pages

### PR DESCRIPTION
**Pull request by Dhivas-Sugumar**
Closes #29 

- The user is redirected to sign in if there is no session data on all pages. Based the solution off of the application portal as recommended. 
- Removed the unstyled signout button at the end. 

The only page an unauthorized user can access.
image

Removed unstyled sign out button. 
image

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -6,17 +6,11 @@ import { theme } from '../styles/theme'
 import { signIn, useSession } from 'next-auth/react'
 import { TextContainer } from '../pageStyles/home.styles'
 import Layout from '../components/layout/Layout'
+import { GetServerSideProps } from 'next';
+import { getServerSideSessionOrRedirect } from '../server/getServerSideSessionOrRedirect';
 
 const Home: NextPage = () => {
   const { data: session } = useSession()
```
Comment (jzhzhang66) [inline pages/index.tsx:13]: Removed unused var

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
NICE JOBBBB!!! Looks great! 

The only comment I have is to remove all unused imports / variables. In emailSender.tsx, index.tsx, and sigmaker.tsx all imports from next-auth/react should be removed since they're not being used

File: pages/index.tsx (hunk 1 above)
Comment (Dhivas-Sugumar) [reply pages/index.tsx]: Removed unused imports.

**Review by jzhzhang66 — APPROVED** [review]
LGTM!!!:)

**Review by mike10911 — APPROVED** [review]
Looks so beautiful! Great job!

**Pull request merged**